### PR TITLE
[#175] Python code refactoring

### LINKED ISSUE #175: Consider cleaning up and formatting python code
# Clarification and motivation

There is now quite a bit of python code in the repository, but it isn't it the best shape.

We should consider cleaning it up, mainly for readability in case someone wants to check it out.

# Acceptance criteria

Revisit the python code and consider:
- making changes for readability and polishing in general
- using a formatting tool, [black](https://github.com/psf/black) seems to be a very common one

### PR DESCRIPTION
**Pull request by rvem**
## Description
There is quite a lot of Python code in the repository now and it doesn't look very well.

This PR provides some refactorings to it:
* Formatting via `black`
* Various refactoring to reduce size of existing modules

## Related issue(s)

Resolves #175

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: docker/package/systemd.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        instances: List[str] = [],
+    ):
+        self.suffix = suffix
+        self.service_file = service_file
+        self.startup_script = startup_script
+        self.startup_script_source = startup_script_source
+        self.prestart_script = prestart_script
+        self.prestart_script_source = prestart_script_source
+        self.config_file = config_file
+        self.instances = instances
```
Comment (ixahedron) [inline docker/package/systemd.py:84]: This looks kinda ugly with the whole giant `__init__` duplication :disappointed: I see that there's an option to use [`@dataclass`](https://docs.python.org/3/library/dataclasses.html), maybe we would want to use that style to automatically assign fields?

**Review by ixahedron — APPROVED** [review]
LGTM aside from a couple of comments. Feel free to decide if any suggestion should be applied at all.

File: docker/package/systemd.py (hunk 1 above)
Comment (rvem) [reply docker/package/systemd.py:84]: :pogchamp:

File: docker/package/model.py [hunk 2]
```diff
@@ -128,34 +108,40 @@ def gen_spec_systemd_part(package):
             service_name = f"%{{name}}-{systemd_unit.suffix}"
         if len(systemd_unit.instances) > 0:
             service_name = f"{service_name}@"
-        install_unit_files += f"install -m 644 {service_name}.service %{{buildroot}}/%{{_unitdir}}\n"
+        install_unit_files += (
+            f"install -m 644 {service_name}.service %{{buildroot}}/%{{_unitdir}}\n"
+        )
```
[the remark this answers is not in the document]
Comment (rvem) [reply docker/package/model.py:113]: AFAIU, that's one way to separate a long line (the other is to use `\`, but black uses parentheses)

File: docker/package/systemd.py (hunk 1 above)
Comment (rvem) [reply docker/package/systemd.py:84]: Sadly, Ubuntu 18.04 (whose docker image is used for package building) has python 3.6.9 and dataclasses aren't supported there. But I think it's time to update and use 20.04 instead

**Pull request merged**